datafusion/core/tests/sqllogictests/test_files/tpch.slt

```diff
@@ -125,7 +125,7 @@ select
     sum(l_quantity) as sum_qty,
     sum(l_extendedprice) as sum_base_price,
     sum(l_extendedprice * (1 - l_discount)) as sum_disc_price,
-    sum(l_extendedprice * (1 - l_discount) * (1 + l_tax)) as sum_charge,
+    sum(cast(l_extendedprice as decimal(12,2)) * (1 - l_discount) * (1 + l_tax)) as sum_charge,
```

Comment:
This decimal calculation actually causes overflow on decimal multiplication because the kernel DataFusion uses doesn't allow precision loss. Previously it accidentally works because we implicitly truncate all input decimal value before multiplying...

Reply:
So I add a cast to get the test result unchanged. To get rid of this `cast`, we can use new multiply kernel at arrow-rs which allows precision-loss.